File: 2018/5/main.py
#!/usr/bin/env python3

x = open('input').read().strip()
def solve(x):
    while True:
        n = len(x)
        for c in 'abcdefghijklmnopqrstuvwxyz':
            x = x.replace(c + c.upper(), '')
            x = x.replace(c.upper() + c, '')
        if len(x) == n:
            break
    return len(x)

print('A:', solve(x))
ans = len(x)
for c in 'abcdefghijklmnopqrstuvwxyz':
    y = x.replace(c, '')
    y = y.replace(c.upper(), '')
    ans = min(ans, solve(y))
print('B:', ans)

File: 2018/5/main.nim
import strutils, sequtils

proc solve(y: string): int =
  var x = y
  var r: seq[(string, string)]
  for c in 'a'..'z':
    r.add((c & c.toUpperASCII, ""))
    r.add((c.toUpperASCII & c, ""))
  while true:
    let n = len(x)
    x = x.multiReplace(r)
    if len(x) == n:
      return len(x)

var x = readFile("input").strip
echo "A: ", solve(x)

var ans = len(x)
for c in 'a'..'z':
  let y = x.replace("" & c, "").replace("" & c.toUpperASCII, "")
  let l = solve(y)
  if l < ans:
    ans = l
echo "B: ", ans
